privacy-protections/https-loop-protection/main.js

```diff
@@ -0,0 +1,142 @@
+const startButton = document.querySelector('#start');
+const downloadButton = document.querySelector('#download');
+
+const testsDiv = document.querySelector('#tests');
+const testsSummaryDiv = document.querySelector('#tests-summary');
+const testsDetailsDiv = document.querySelector('#tests-details');
+
+const TEST_DOMAIN = 'good.third-party.site';
+
+const tests = [
+    {
+        id: 'upgrade-navigation',
+        run: () => {
```

Comment:
most of this file, besides the contents of this test, is just a template. We should probably extract it to some reusable helpers instead of copying it over and over